lidder_test.go

```diff
@@ -37,18 +40,70 @@ rules:
       - file_a.go
       - file_b.go`
 
-	d, err := parse([]byte(conf))
+	return parse([]byte(conf))
+}
+
+func (s *Zuite) TestParseConfiguration() {
+	d, err := configFile()
 	require.NoError(s.T(), err)
 
-	require.Equal(s.T(), []string{"^.*\\.go$"}, d.Include)
-	require.Equal(s.T(), []string{"^.*\\bvendor/.*$"}, d.Exclude)
+	require.Equal(s.T(), []string{"^abc/.*\\.go$", "^def/.*\\.go$"}, d.Include)
+	require.Equal(s.T(), []string{"^.*\\bvendor/.*$", "^.*_test\\.go$"}, d.Exclude)
 	require.Equal(s.T(), 1, len(d.Rules))
 	for _, rule := range d.Rules {
 		require.Equal(s.T(), "panic\\(", rule.Pattern)
 		require.Equal(s.T(), []string{"file_a.go", "file_b.go"}, rule.Expected)
 	}
 }
 
+func (s *Zuite) TestShouldCheck() {
+	d, err := configFile()
+	require.NoError(s.T(), err)
+
+	require.False(s.T(), d.shouldCheck("abc/hello_test.go"))
+	require.True(s.T(), d.shouldCheck("abc/hello.go"))
+	require.True(s.T(), d.shouldCheck("def/goodbye.go"))
+	require.False(s.T(), d.shouldCheck("abcdef/goodbye.go"))
+}
+
+func (s *Zuite) TestMatchAgainstLine() {
+	d, err := configFile()
+	require.NoError(s.T(), err)
+
+	testLines := []string{
+		"if blah == blahblah {",
+		"    panic(\"whoa\")",
+		"}",
+	}
+
+	for _, rule := range d.Rules {
```

Comment:
This doesn't feel necessary: you're checking the base state post `parse`, so this feels like it is really testing `parse`, not `matchAgainstLine`.

Reply:
it's true that testing matchAgainstLine would be pretty trivial, but i don't think this is testing just parse. there are no other tests checking that rule.actualFilenames is populating correctly and that rule.Mismatches is giving us what we want. perhaps the test should be named something else?

i think the name and the sample code definitely distract from what is actually being tested

Reply:
I've renamed it TestRuleMatching, let me know what your thoughts are!